backend/de.metas.business/src/main/java/de/metas/costrevaluation/impl/CostRevaluationDAO.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * #%L
+ * de.metas.business
+ * %%
+ * Copyright (C) 2022 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.costrevaluation.impl;
+
+
+import de.metas.costrevaluation.ICostRevaluationDAO;
+import de.metas.document.DocTypeId;
+import de.metas.document.DocTypeQuery;
+import de.metas.document.IDocTypeDAO;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.ad.dao.impl.ActiveRecordQueryFilter;
+import org.compiere.model.I_M_CostRevaluationLine;
+import org.compiere.model.X_C_DocType;
+import org.compiere.util.Env;
+
+import java.util.List;
+
+public class CostRevaluationDAO  implements ICostRevaluationDAO
+{
+
+	private final IQueryBL queryBL = Services.get(IQueryBL.class);
+	private final IDocTypeDAO docTypeDAO  = Services.get(IDocTypeDAO.class);
+
+	@Override
+	public List<I_M_CostRevaluationLine> retrieveLinesByCostRevaluationId(@NonNull final CostRevaluationId costRevaluationId)
+	{
+		return queryBL.createQueryBuilder(I_M_CostRevaluationLine.class)
+				.addEqualsFilter(I_M_CostRevaluationLine.COLUMN_M_CostRevaluation_ID, costRevaluationId)
+				.filter(ActiveRecordQueryFilter.getInstance(I_M_CostRevaluationLine.class))
+				.orderBy(I_M_CostRevaluationLine.COLUMN_M_CostRevaluationLine_ID)
+				.create()
+				.list();
+	}
+
+
+	@Override
+	public DocTypeId retrieveCostRevaluationDocTypeId()
+	{
+		final DocTypeQuery docTypeQuery = DocTypeQuery.builder()
+				.docBaseType(X_C_DocType.DOCBASETYPE_CostRevaluation)
+				.docSubType(DocTypeQuery.DOCSUBTYPE_Any)
+				.adClientId(Env.getClientId().getRepoId())
+				.adOrgId(Env.getOrgId().getRepoId())
+				.build();
+
+		return docTypeDAO.getDocTypeIdOrNull(docTypeQuery);
```

Comment:
calling a DAO from another DAO it's a sign of bad design.

In this case, IMHO, if we need `retrieveCostRevaluationDocTypeId`, i suggest to:
* introduce CostRevaluationService (spring component) and move it there
* convert this DAO to CostRevaluationRepository and allow only CRUD operations which are related to M_CostRevaluation and M_CostRevaluationLine

about `retrieveCostRevaluationDocTypeId` which returns null, i suggest
* either return Optional<DocTypeId>
* or return non-null DocTypeId